xarray/core/missing.py

```diff
@@ -30,35 +30,30 @@ def __repr__(self):
 
 
 class NumpyInterpolator(BaseInterpolator):
-    '''One-dimensional linear interpolation.
+    """One-dimensional linear interpolation.
 
     See Also
     --------
     numpy.interp
-    '''
-
-    def __init__(self, xi, yi, method='linear', fill_value=None, **kwargs):
+    """
+    def __init__(self, xi, yi, method='linear', fill_value=None, period=None):
 
         if method != 'linear':
             raise ValueError(
                 'only method `linear` is valid for the NumpyInterpolator')
 
         self.method = method
         self.f = np.interp
-        self.cons_kwargs = kwargs
-        self.call_kwargs = {'period': self.cons_kwargs.pop('period', None)}
+        self.cons_kwargs = {}
```

Comment:
Should still be `kwargs`

Reply:
No, because it's always empty after you remove period

Reply:
yes, indeed